lib/appcanary/configuration.rb

```diff
@@ -4,18 +4,46 @@ module Appcanary
   APPCANARY_DEFAULT_BASE_URI = "https://appcanary.com/api/v3"
 
   class Configuration
-    attr_accessor :base_uri, :api_token, :monitor_name
+    attr_accessor :base_uri, :api_key, :monitor_name, :gemfile_lock
 
     def initialize
       self.base_uri = APPCANARY_DEFAULT_BASE_URI
+      self.monitor_name = maybe_guess_monitor_name
+      self.gemfile_lock = locate_gemfile_lock
+    end
+
+    def maybe_guess_monitor_name
+      Rails.application.class.parent_name if defined?(Rails)
+    end
+
+    def locate_gemfile_lock
+      if defined?(Bundler)
+        Bundler.default_lockfile
+      else
+        begin
+          require "bundler"
+          locate_gemfile_lock
+        rescue LoadError
+          # ignore, handle at resolution time
+          return
+        end
+      end
     end
 
     def valid?
```

Comment:
seems like this is checking for whether static conf was provided, and not "validity" per se. Rename?